CHANGELOG.md

```diff
@@ -47,6 +47,14 @@
 * Add methods from SE-0348 to `UnusedDeclarationRule`.  
   [JP Simard](https://github.com/jpims)
 
+* Show warnings in the console for
+  * opt-in rules that are listed in the `disabled_rules` configuration section,
```

Comment:
How does this work with nested configurations? Like if an opt-in rule is enabled at the root configuration and disabled in a nested configuration?

Reply:
Hm ... I guess it does not work at all. I did not think about nested configurations to be honest. Considering them now, my assumptions (no need to disable opt-in rules, no need to enable rules active by default) are not valid anymore. It makes sense to disable an opt-in rule if it was enabled in the parent configuration for example.

It would still be reasonable to validate merged configurations. But how to report warnings then? How to tell to which combination of configurations a warning belongs to?

The initial idea for these checks came from the fact that Analyzer rules were specified but were not considered because `swiftlint lint` was run instead of `swiftlint analyze`. It doesn't even matter whether they are listed in `analyzer_rules` or `opt_in_rules`. The important thing is to actually run SwiftLint in `analyze` mode.

Reflecting on all this, I think there is not much we can do to help the user. Analyzer rules must be triggered specially. SwiftLint cannot decide whether it is running in the right mode just by the configuration. And it also doesn't hurt to have rules listed redundantly or in the "wrong" section.

The only thing that could help is providing a hint like "`rule_id` as an Analyzer rule belongs into the `analyzer_rules` section to run it in the special `analyze` mode". Feels a bit over-engineered now, however. If all rules are listed in the correct section, there won't be a warning and you can still forget to run `swiftlint analyze` separately.

That said, it's probably better to skip implementation of these warnings entirely and close the PR. 😅

Reply:
Yes, sorry to say but I also think we should just close the PR.

Reply:
Fine for me. Good you asked the question. Better than adding something unhelpful, confusing or even wrong.

At least, we though about it and know better now and for the future.